### PR TITLE
[Mellanox] Add support for SN4600 system

### PR DESCRIPTION
**Pull request by DavidZagury**
**- Why I did it**
Add support for new 64x200G SN4600 systems

**- How I did it**
Added all relevant files (w/o platform.json and hwsku.json as they will come later) with default SKU.

**- How to verify it**
Install image on switch, verify all ports are up and configured properly, run full platform SONiC tests.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn4600-r0/sensors.conf [hunk 1]
```diff
@@ -0,0 +1,189 @@
+################################################################################
+# Copyright (c) 2021 Mellanox Technologies
+#
+# Platform specific sensors config for SN4600C
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-mlnx_msn4600-r0/sensors.conf:4]: please fix system name. it is 4600 and not 4660c

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py [hunk 2]
```diff
@@ -128,9 +128,13 @@
...
 
-platform_dict_thermal = {'x86_64-mlnx_msn2700-r0': 0, 'x86_64-mlnx_lssn2700-r0':0, 'x86_64-mlnx_msn2740-r0': 3, 'x86_64-mlnx_msn2100-r0': 1, 'x86_64-mlnx_msn2410-r0': 2, 'x86_64-mlnx_msn2010-r0': 4, 'x86_64-mlnx_msn3420-r0':9, 'x86_64-mlnx_msn3700-r0': 5, 'x86_64-mlnx_msn3700c-r0': 6, 'x86_64-mlnx_msn3800-r0': 7, 'x86_64-mlnx_msn4600c-r0':9, 'x86_64-mlnx_msn4700-r0': 8, 'x86_64-mlnx_msn4410-r0': 8}
+platform_dict_thermal = {'x86_64-mlnx_msn2700-r0': 0, 'x86_64-mlnx_lssn2700-r0': 0, 'x86_64-mlnx_msn2740-r0': 3,
+                         'x86_64-mlnx_msn2100-r0': 1, 'x86_64-mlnx_msn2410-r0': 2, 'x86_64-mlnx_msn2010-r0': 4,
+                         'x86_64-mlnx_msn3420-r0': 9, 'x86_64-mlnx_msn3700-r0': 5, 'x86_64-mlnx_msn3700c-r0': 6,
+                         'x86_64-mlnx_msn3800-r0': 7, 'x86_64-mlnx_msn4600-r0': 12, 'x86_64-mlnx_msn4600c-r0': 9,
+                         'x86_64-mlnx_msn4700-r0': 8, 'x86_64-mlnx_msn4410-r0': 8}
 thermal_profile_list = [
-    # 2700
+    # 0 2700
```
Comment (liat-grozovik) [inline platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py:137]: if the number you added is index to the table then there is something strange
you added 4600 as 12 but in this table it is not the last one it was added in between supported platforms

**Review by liat-grozovik — CHANGES_REQUESTED** [review]

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py (hunk 2 above)
Comment (DavidZagury) [reply platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py:137]: The numbers I added are for the index in thermal_profile_list, which in there 4600 is last.
Those numbers need to be the same as the values in platform_dict_thermal, in which the order does not matter.

**Review by liat-grozovik — APPROVED** [review]

File: device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/port_config.ini [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# name          lanes               alias     index
+Ethernet0       0,1,2,3             etp1      1
+Ethernet8       8,9,10,11           etp2      2
```
Comment (prsunny) [inline device/mellanox/x86_64-mlnx_msn4600-r0/ACS-MSN4600/port_config.ini:3]: why is this a difference of 8.. its 4 lanes.. @liat-grozovik , @DavidZagury, could you please correct this?

**Pull request merged**